draft-ietf-webtrans-overview.md

```diff
@@ -367,10 +367,22 @@ receive side aborted
 : Indicates that the peer has aborted the corresponding send side of the
   stream.  An unsigned 8-bit error code from the peer may be available.
 
-Data Recvd state reached
-: Indicates that no further data will be transmitted or retransmitted on the
-  local send side, and that the FIN has been sent.  Data Recvd implies that
-  aborting send-side is a no-op.
+all data committed
+: Indicates that all of the outgoing data on the stream, including the end stream indication, is
+  in the state where aborting the send side would have no further effect on any
+  data being delivered.  
+  
+: For protocols, like HTTP/2, stream data might be passed to another
+  component (like a kernel) for transmission. Once data is passed to that
+  component it might not be possible to abort the sending of stream data
+  without also aborting the entire connection.
+  For these protocols, data is considered committed once it passes to the
+  other component.
+  
+: A protocol, like HTTP/3, that uses a more integrated stack might be able to
+  retract data further into the process. For these protocols, sending on a
+  stream might be aborted at any time until all data has been received,
+  corresponding to the "Data Recvd" state in QUIC; see {{Section X.Y of QUIC}}.
```

Comment:
"Data Recvd" indicates that ACKs have been received that cover all data sent. Do all QUIC stacks expose this to the application layer?  HTTP/3 made a hard line that the application layer didn't have or need access to transport-level ACKs of its data.

Reply:
Due to the way W3C APIs are structured, we have to keep a stream in a "kind of open" state so that the application could send a reset for a stream in the window between the FIN being sent and all data being acknowledged.  It's somewhat unfortunate, but it isn't really meant to be a solid signal for the application (in fact, I think in WebTransport over HTTP/2, this will be probably signaled instantly).